### PR TITLE
[liveleak] improve

### LINKED ISSUE #6542: [LiveLeak] Extractor only downloads first video from a multi-video page
- **URL:** http://www.liveleak.com/view?i=677_1439397581
- **Platform:** openSUSE 13.1, x86_64, XFCE
- **Version:** 2015.08.09

The page contains 3 videos, but only the first one is grabbed. Using `--force-generic-extractor` makes it pick up all three videos, but shouldn't be necessary.

```
sven@linux-etoq:~/archive/tianjin-explosion> youtube-dl -v http://www.liveleak.com/view?i=677_1439397581
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'http://www.liveleak.com/view?i=677_1439397581']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.08.09
[debug] Python version 2.7.6 - Linux-3.11.10-29-desktop-x86_64-with-SuSE-13.1-x86_64
[debug] exe versions: ffmpeg 2.7.2, ffprobe 2.7.2, rtmpdump 2.4
[debug] Proxy map: {}
[LiveLeak] 677_1439397581: Downloading webpage
[debug] Invoking downloader on u'https://cdn.liveleak.com/80281E/ll_a_s/2015/Aug/12/LiveLeak-dot-com-677_1439397581-000Khy0Ujx06UAlXOv04010401008hMd0k01_1439397605.mp4?d5e8cc8eccfb6039332f41f6249e92b06c91b4db65f5e99818bdd49f4944d2d32701&ec_rate=230'
[download] Fuel Depot in China Explosion caught on video-677_1439397581.mp4 has already been downloaded
[download] 100% of 1.88MiB
```

### LINKED ISSUE #7506: [LiveLeak] Embedded YouTube videos are not detected
- **URLs:** http://www.liveleak.com/view?i=993_1447462861, http://www.liveleak.com/view?i=cef_1447457924
- **Platform:** openSUSE 13.1, x86_64, XFCE
- **Version:** 2015.11.13

As issue title says. Issue occurs for both URLs above, only the output for one of them (using an alias) is included below.

```
sven@linux-etoq:/mnt/storage2/archive/paris-attacks-youtube/liveleak> ytdl -v http://www.liveleak.com/view?i=993_1447462861
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'--title', u'--continue', u'--retries', u'4', u'--write-info-json', u'--write-description', u'--write-thumbnail', u'--write-annotations', u'--all-subs', u'--ignore-errors', u'--merge-output-format', u'mkv', u'-f', u'bestvideo+bestaudio/best', u'-v', u'http://www.liveleak.com/view?i=993_1447462861']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2015.11.13
[debug] Python version 2.7.6 - Linux-3.11.10-29-desktop-x86_64-with-SuSE-13.1-x86_64
[debug] exe versions: ffmpeg 2.8.2, ffprobe 2.8.2, rtmpdump 2.4
[debug] Proxy map: {}
[LiveLeak] 993_1447462861: Downloading webpage
ERROR: Unable to extract embed URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 661, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python2.7/site-packages/youtube_dl/extractor/common.py", line 290, in extract
    return self._real_extract(url)
  File "/usr/lib/python2.7/site-packages/youtube_dl/extractor/liveleak.py", line 79, in _real_extract
    webpage, 'embed URL')
  File "/usr/lib/python2.7/site-packages/youtube_dl/extractor/common.py", line 584, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
RegexNotFoundError: Unable to extract embed URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

### PR DESCRIPTION
**Pull request by aeph6Ee0**
- extracts more than just one video per page (fix  #6542)
- extracts embedded youtube videos (fix #7506)
- removes the `&ec_rate=...` rate limiting parameter from the urls, to download videos with full speed

### CODE REVIEW

**Comment by dstftw** [comment]
Add tests for each new scenario.

File: youtube_dl/extractor/liveleak.py [hunk 1]
```diff
@@ -65,55 +84,96 @@ def _real_extract(self, url):
...
+                'description': video_description,
+                'uploader': video_uploader,
+                'formats': formats,
+                'age_limit': age_limit,
+            })
+
+        # collect embedded videos:
+        embed_urls = list()
+
+        # prochan.com:
```
Comment (dstftw) [inline youtube_dl/extractor/liveleak.py]: Avoid senseless and obvious comments.

File: youtube_dl/extractor/liveleak.py [hunk 2]
```diff
@@ -65,55 +106,93 @@ def _real_extract(self, url):
...
+                'age_limit': age_limit,
+            })
+
+        if len(entries) == 1:
+            return entries[0]
+        else:
+            return {
+                '_type': 'multi_video',
+                'id': page_id,
+                'title': 'Fuel Depot in China Explosion caught on video',
```
Comment (dstftw) [inline youtube_dl/extractor/liveleak.py]: Why is this hardcoded?

File: youtube_dl/extractor/liveleak.py [hunk 3]
```diff
@@ -65,55 +106,93 @@ def _real_extract(self, url):
...
+                '_type': 'url_transparent',
+                'id': page_id + self._video_count(entries),
+                'url': embed_url,
+                'title': video_title,
+                'description': video_description,
+                'uploader': video_uploader,
+                'age_limit': age_limit,
+            })
+
+        if len(entries) == 1:
```
Comment (dstftw) [inline youtube_dl/extractor/liveleak.py]: Just return the playlist all the time.

File: youtube_dl/extractor/liveleak.py [hunk 4]
```diff
@@ -65,55 +106,93 @@ def _real_extract(self, url):
...
+
+        embed_urls = list()
+
+        for embed_prochan in re.findall(
+                r'<iframe[^>]+src="(http://www.prochan.com/embed\?[^"]+)"',
+                webpage):
+            embed_urls.append(embed_prochan)
+
+        for embed_youtube in re.findall(
+                r'<iframe[^>]+src="(http[s]?://www.youtube.com/embed/[^"]+)"',
```
Comment (dstftw) [inline youtube_dl/extractor/liveleak.py]: This can be simplified to `https?://www\.youtube\.com/embed/[^"]+)`.

**Comment by aeph6Ee0** [comment]
I fixed all complaints.

**Comment by yan12125** [comment]
liveleak.py is completely rewritten due to website redesign. (#12053) Would you like to update this patch?

**Comment by aeph6Ee0** [comment]
I'll leave this to others.

**Pull request closed without merging**